lib/spack/spack/bootstrap.py

```diff
@@ -516,11 +535,29 @@ def ensure_bootstrap_configuration():
                         # We may need to compile code from sources, so ensure we have
                         # compilers for the current platform before switching parts.
                         _add_compilers_if_missing()
+                        spack.config.set('bootstrap', user_configuration['bootstrap'])
+                        spack.config.set('config', user_configuration['config'])
                         with spack.modules.disable_modules():
                             with spack_python_interpreter():
                                 yield
 
 
+def _read_and_sanitize_configuration():
+    """Read the user configuration that needs to be reused for bootstrapping
+    and remove the entries that should not be copied over.
+    """
+    # Read config.yaml but pop the install tree (the entry will not be considered
```

Comment:
config.yaml is a confusing comment here, you mean: read the effective/merged config?

Reply:
Yeah, it's the merged config from whatever setup a user has. Let me use "section" in the comment if that's clearer.

Reply:
the config config section ;p rolls of the tongue

Reply:
Done in 60f0837